File: python/dnd-character/dnd_character.py
import random
import math

class Character:
    def __init__(self):
        self.strength = self.ability()
        self.dexterity = self.ability()
        self.constitution = self.ability()
        self.intelligence = self.ability()
        self.wisdom = self.ability()
        self.charisma = self.ability()
        self.hitpoints = 10 + modifier(self.constitution)


    def ability(self):
        rolls = []
        for i in range(4):
            rolls.append(random.randint(1,6))
        rolls.remove(min(rolls))
        return sum(rolls)

def modifier(val):
    return math.floor((val - 10) / 2)

File: ruby/dnd-character/dnd_character.rb
class DndCharacter
  attr_reader :constitution, :strength, :dexterity, :intelligence, :wisdom, :charisma, :hitpoints

  def self.modifier(num)
    ((num - 10)/ 2).floor
  end

  def initialize
    @constitution = rando
    @hitpoints = 10 + DndCharacter.modifier(@constitution)
    @strength = rando
    @dexterity = rando
    @intelligence = rando
    @wisdom = rando
    @charisma = rando
  end

  def rando
    rolls = ((1..4).map { rand(5) + 1 })
    rolls.delete_at(rolls.index(rolls.min))
    rolls.sum
  end

end
